### ISSUE TITLE
Crash at closing

### ISSUE THREAD

**Issue by paramecie** [rgriebl/brickstore#305]
Just got this, no idea why : BrickStore was opened for some time, I gave it focus back, closed it and immediately got the message:

![image](https://user-images.githubusercontent.com/32078594/111377449-f3e9b080-86a0-11eb-9fd2-296ce91c7a5f.png)

**Build 354**

First time it (ever) happens to BS.
I'll try to get more diagnostic info if it happens again...

**Comment by paramecie** [comment]
Mmmmh, does BS have opened and NOT locked files in temp folder (usually c:\windows\temp - but not here)?

**Comment by rgriebl** [comment]
The autosaves (in case BS crashes) are saved to Windows' Temp folder. The standard for Win10 would be "C:/Users/<USER>/AppData/Local/Temp".
There are no locks however. It's just a background thread that dumps the current document every minute (if it's been modified)

**Comment by paramecie** [comment]
Win 7 here Robert :)

Then it's the cause of the problem.

Windows / Windows apps have a tendancy to accumulate Temp files, and users usually don't see them - you end up with thousands of trash files not trashed.

So here I've a batch that cleans this folder from time to time. That means all not locked files are deleted. If BrickStore is expecting a file, this may be a problem if they're not locked....

**Comment by paramecie** [comment]
Solutions could be:

* ignore the problem ;-)

* keep the files opened or lock in any way

* keep them in BrickStore cache in some specific folder - not the windows trashy temp folder

**Comment by rgriebl** [comment]
No - this shouldn't affect BS at all

**Comment by paramecie** [comment]
Ah sorry. Then I've no clue.

**Comment by paramecie** [comment]
Is it possible it crashed because my cleaning task removed the file at the exact moment the backup file was saved ?
Or that during the closing it was supposed to clear the temp files, but because they didn't exist it crashed?

Is it possible it crashed outside a try/catch, which provoked the Runtime Lib to crash, hence the message above?
Maybe there are those places without try/catch when the app closes?

Just ideas... ;)

**Comment by paramecie** [comment]
Didn't encounter any crash since this day.

When I think at it, at that moment of the crash I closed many things (as I do often having finished a task, and switching to something else).

So it *is* possible that **another** app (using VC Redist also) crashed but with some sort of delay, making the message appear just after I closed BrickStore.

In short maybe the message didn't come from BrickStore.

As for now I didn't have any problem, I can't confirm anything, but I thought it's important to tell you...
